File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max_ = *max_element(candies.begin(), candies.end()); 
        vector<bool> answer(candies.size(), false); 

        for(int i = 0; i < candies.size(); i++){
            int curr = candies[i] + extraCandies; 
            if(curr >= max_) answer[i] = true; 
        }

        return answer; 
    }
};

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> answer = new ArrayList<>(Collections.nCopies(candies.length,false)); 
        int max_ = 0; 
        for(int n : candies) max_ = Math.max(max_,n); 
        for(int i = 0; i < candies.length; i++){
            if(candies[i] + extraCandies >= max_){
                answer.set(i,true); 
            }
        }


        return answer; 
    }
}